internal/deb/extract_test.go

```diff
@@ -352,6 +352,40 @@ var extractTests = []extractTest{{
 		},
 	},
 	error: `cannot extract from package "test-package": path /dir/ requested twice with diverging mode: 0777 != 0000`,
+}, {
+	summary: "Hard link must be created if specified in the tarball",
+	pkgdata: testutil.MustMakeDeb([]testutil.TarEntry{
+		testutil.Dir(0755, "./"),
+		testutil.Reg(0644, "./file1.txt", "text for file1.txt"),
+		testutil.Hln(0644, "./file2.txt", "./file1.txt"),
+	}),
+	options: deb.ExtractOptions{
+		Extract: map[string][]deb.ExtractInfo{
+			"/*.txt": []deb.ExtractInfo{{
+				Path: "/*.txt",
+			}},
+		},
+	},
+	result: map[string]string{
+		"/file1.txt": "file 0644 e926a7fb",
+		"/file2.txt": "file 0644 e926a7fb",
+	},
+	notCreated: []string{},
+}, {
+	summary: "Dangling hard link must raise an error",
+	pkgdata: testutil.MustMakeDeb([]testutil.TarEntry{
+		testutil.Dir(0755, "./"),
+		// testutil.Reg(0644, "./file1.txt", "text for file1.txt"),
+		testutil.Hln(0644, "./file2.txt", "./file1.txt"),
+	}),
+	options: deb.ExtractOptions{
+		Extract: map[string][]deb.ExtractInfo{
+			"/*.txt": []deb.ExtractInfo{{
+				Path: "/*.txt",
+			}},
+		},
+	},
+	error: `cannot extract from package "test-package": the target file does not exist: .*/file1.txt`,
```

Comment:
In error messages we normally do not use `.*` because we want to see the error as the user would see it.

Reply:
This is to match the path prefix of the target file, since in the tests the target file with absolute paths is printed like `/tmp/check-4089915458/20/file1.txt`. Without knowing the path prefix in the test case (or maybe we can get it somewhere?), it seems inevitable to use a `.*` here.